rec-YnH
==========

Table of Contents:

- [Summary](#summary)
- [Requirements](#requirements)
- [Installation](#installation)
	- [rec-YnH](#vast-tools-1)
- [Usage](#usage)
	- [Help](#help)
	- [Quick Usage](#quick-usage)
	- [Alignment](#align)
	- [Interaction Score](#score)
	- [Merge Interaction Scores](#merge)
- [Issues](#issues)
- [Contributions](#contributions)
- [Citation](#citation)
- [References](#references)
	
Summary
-------
[What is recYnH?]

This program is taking recYnH sequencing files and generating recYnH interaction score matrix to correponding genes.

We assumed that the sequencing format is the same as we described in [Nature Communication 2018 Paper].

 After setting proper parameter in recYnH.py file

 User can run the program as following command

~~~~
usage: recYnH.py [-h] COMMAND ...
~~~~

~~~~
recYnH program

positional arguments:
  COMMAND     sub-command help
    align     a help for align
    score     a help for score
    merge     a help for merge

optional arguments:
  -h, --help  show this help message and exit

Commands:
  align      Align the FASTQ sequencing files into bait and prey sequences to generate interaction matrix
  score      Usging two interaction matries to generate an interaction score matrix
  merge      Merge several interaction score matries and apply quartile correction to generate a final average interaction score matrix

Run 'recYnH.py COMMAND --help' for more information on a command.
~~~~

Requirements
------------

rec-YnH requires the following software:
 * python 2.7 or higher, with the following packages installed (see Installation Section):
   * scipy, https://pypi.org/project/scipy/
 * cutadapt, http://cutadapt.readthedocs.io/en/stable/installation.html
 * blast+ 2.6.0 or higher, http://ftp.ncbi.nlm.nih.gov/blast/executables/blast+
 
 * R 3.1 or higher, with the following packages installed (see Installation Section):
   * RColorBrewer
   * pheatmap
   * mixtools

Installation
------------

~~~~
> git clone https://github.com/lionking0000/recYnH.git
~~~~

R packages

RUN Rscript -e "install.packages(c('pheatmap','RColorBrewer','mixtools'))"

Alignment
------------

Here is the example of how to do alignment (mapping) FASTQ files into reference sequences (FASTA).

command: align

i1: FASTA file for baits

i2: FASTA file for preys

f1: FASTQ file for read1 (baits)

f2: FASTQ file for read2 (preys)

o: output folder

n: output name

Example 1
~~~~
python recYnH.py align -i1 ./example/db/roth2016_control_set_plus_control.fa -f1 ./example/fastq/EXP1_W_R1.25000.fastq -f2 ./example/fastq/EXP1_W_R2.25000.fastq -o ./example/output/ -n EXP1_W.25k
~~~~
~~~~
python recYnH.py align -i1 ./example/db/roth2016_control_set_plus_control.fa -f1 ./example/fastq/EXP1_Q_R1.25000.fastq -f2 ./example/fastq/EXP1_Q_R2.25000.fastq -o ./example/output/ -n EXP1_Q.25k
~~~~


Example 2
~~~~
python recYnH.py align -i1 ./example/db/roth2016_control_set_plus_control.fa -f1 ./example/fastq/EXP2_W_R1.25000.fastq -f2 ./example/fastq/EXP2_W_R2.25000.fastq -o ./example/output/ -n EXP2_W.25k
~~~~

~~~~
python recYnH.py align -i1 ./example/db/roth2016_control_set_plus_control.fa -f1 ./example/fastq/EXP2_Q_R1.25000.fastq -f2 ./example/fastq/EXP2_Q_R2.25000.fastq -o ./example/output/ -n EXP2_Q.25k
~~~~


Interaction Score 
------------

Using two matrix files to generate Interaction Score matrix

command: score

m1 : background matrix (Recombination Selection)

m2 : interaction matrix (Recombination Interaction Selection)

o: output folder

n: output name


Example 1
~~~~
python recYnH.py score -m1 ./example/output/EXP1_W.25k -m2 ./example/output/EXP1_Q.25k -o ./example/output/ -n EXP1.25k
~~~~


Example 2
~~~~
python recYnH.py score -m1 ./example/output/EXP2_W.25k -m2 ./example/output/EXP2_Q.25k -o ./example/output/ -n EXP2.25k
~~~~


Merge Interaction Scores and apply upper quartile correction
------------

Merge several interaction score matrices and apply quartile correction.

command: merge

i: one or more interaction score (IS) matrices (generated by "score" command)

q: percentile value used for correction (default = 0.75; upper quartile)

o: output folder

n: output name


Example 1
~~~~
python recYnH.py merge -i ./example/output/EXP1.25k.nis.txt ./example/output/EXP2.25k.nis.txt ./example/output/EXP3.25k.nis.txt -o ./example/output/ -n EXP.25k.avgIS
~~~~

